### ISSUE TITLE
Use first super() argument as the target type

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#3926]
Currently mypy ignores the type that the first `super()` argument refers to -- it assumes that it always refers to the enclosing class. This is incorrect. For example, here mypy generates a false negative: 

```py
class A:
    def f(self) -> None: pass

class B(A):
    def g(self) -> None:
        super(A, self).f()  # No error though this will fail at runtime

B().g()  # Runtime error
```

Once we have this, it should be easy to also support the 2-argument form of `super()` outside class definitions.

This is follow-up to #3919.

**Comment by ilevkivskyi** [comment]
Now mypy correctly gives an error here.
